app/DoctrineMigrations/Version20161122203647.php

```diff
@@ -32,26 +32,32 @@ public function setContainer(ContainerInterface $container = null)
 
     private function getTable($tableName)
     {
-        return $this->container->getParameter('database_table_prefix') . $tableName;
+        return $this->container->getParameter('database_table_prefix').$tableName;
     }
 
     /**
      * @param Schema $schema
      */
     public function up(Schema $schema)
     {
-        $this->abortIf($this->connection->getDatabasePlatform()->getName() === 'sqlite', 'This up migration can\'t be executed on SQLite databases, because SQLite don\'t support DROP COLUMN.');
+        $userTable = $schema->getTable($this->getTable('user'));
 
-        $this->addSql('ALTER TABLE '.$this->getTable('user').' DROP expired');
-        $this->addSql('ALTER TABLE '.$this->getTable('user').' DROP credentials_expired');
+        $this->skipIf(false === $userTable->hasColumn('expired'), 'It seems that you already played this migration.');
```

Comment:
Won't this skip the whole migration even if `credentials_expired` wasn't removed?